mutation_partition.cc

```diff
@@ -2081,10 +2084,12 @@ to_data_query_result(const reconcilable_result& r, schema_ptr s, const query::pa
             if (res.stop == stop_iteration::yes) {
                 break;
             }
+            co_await coroutine::maybe_yield();
         }
     } else {
         for (const partition& p : r.partitions()) {
-            const auto res = p.mut().unfreeze(s).consume(consumer, reverse);
+            auto m = co_await p.mut().unfreeze_gently(s);
+            const auto res = std::move(m).consume(consumer, reverse);
```

Comment:
Maybe we need a gentle variant of `mutation::consume()` too. It can be easily implemented with the help of the pause-resume support.

Reply:
I see that is exactly what you did.

Reply:
This was added in https://github.com/scylladb/scylla/pull/10482/commits/5ec80274a7a6847bf9fac7946f448d2f134e688a